src/diagnostic/index.ts

```diff
@@ -0,0 +1,133 @@
+import path from 'path';
+
+import semver from 'semver';
+import vscode from 'vscode';
+
+import { configuration, updateConfiguration } from '../configuration';
+import { commands } from '../utils/constants';
+import { findPkgInstallDir } from '../utils/pkg';
+import { getPackageInfo } from '../utils/pkg-info';
+import { detectPm } from '../utils/pm';
+
+export const diagnosticCollection = vscode.languages.createDiagnosticCollection('lints');
+
```

Comment:
Maybe better to prefix with extension name, so user can know the warning is from our extension